### PR TITLE
remove ethers dependency for alloy

### PR DESCRIPTION
**Pull request by tcoratger**
Time spent on this PR:

Resolves: #<Issue number>

# Pull Request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Build-related changes
- [ ] Documentation content changes
- [ ] Testing

# What is the new behavior?

# Does this introduce a breaking change?

- [ ] Yes
- [ ] No

### CODE REVIEW

File: src/eth_provider/contracts/erc20.rs [hunk 1]
```diff
@@ -34,17 +31,29 @@ impl<P: EthereumProvider> EthereumErc20<P> {
...
     pub async fn balance_of(self, evm_address: Address, block_id: BlockId) -> EthProviderResult<U256> {
-        // Prepare the calldata for the bytecode function call
-        let address = EthersAddress::from_slice(evm_address.as_slice());
-        let calldata = IERC20Calls::BalanceOf(BalanceOfCall { account: address }).encode();
+        // Prepare the calldata for the bytecode function call.
+        // Create a provider using the `ProviderBuilder`` and the RPC configuration from the environment.
+        let provider = ProviderBuilder::new()
+            .on_builtin(&RPCConfig::from_env().expect("Failed to load Kakarot RPC config").socket_addr)
+            .await
+            .expect("Failed to create provider via alloy `ProviderBuilder`");
```
Comment (greged93) [inline src/eth_provider/contracts/erc20.rs:39]: Is there no way to get the calldata without using a false provider?

File: tests/tests/alchemy_api.rs [hunk 2]
```diff
@@ -29,17 +29,14 @@ async fn test_token_balances(#[future] erc20: (Katana, KakarotEvmContract), _set
...
 
-    eoa.call_evm_contract(
-        &erc20,
-        "mint",
-        (Token::Address(to), Token::Uint(ethers::abi::Uint::from_big_endian(&amount.to_be_bytes::<32>()[..]))),
-        0,
-    )
-    .await
-    .expect("Failed to mint ERC20 tokens");
+    println!("Address: {:?}", eoa.evm_address().unwrap().as_slice());
```
Comment (greged93) [inline tests/tests/alchemy_api.rs:35]: leftover

**Review by greged93 — COMMENTED** [review]
very nice to be able to remove ethers!

File: tests/tests/alchemy_api.rs (hunk 2 above)
Comment (tcoratger) [reply tests/tests/alchemy_api.rs]: Thanks, fixed

File: src/eth_provider/contracts/erc20.rs (hunk 1 above)
Comment (tcoratger) [reply src/eth_provider/contracts/erc20.rs]: Fixed, I found a much better solution with simply `abi_encode`

**Review by greged93 — APPROVED** [review]

**Pull request merged**